internal/cmd/shell.go

```diff
@@ -85,7 +85,9 @@ func shell(cmd *cobra.Command, cmdArgs []string) error {
 		args = append(args, strings.Join(cmdArgs[1:], " "))
 	} else {
 		// Bring user into shell if no command is specified.
-		args = append(args, "exec $(getent passwd $(whoami) | awk -F: '{ print $7 }')")
+		shell := "$(getent passwd $(id -u) | cut -d: -f 7)"
+		name := "-$(basename " + shell + ")"
+		args = append(args, fmt.Sprintf("exec -a %q %q", name, shell))
```

Comment:
just saying, I've been burned by `%q` before...

Reply:
but I assume that was user error on my part 😆

Reply:
Haha it might be, but it seems to work alright? https://play.golang.org/p/iHJ_QWLS_Tx